modules/rosa-basic-concepts.adoc

```diff
@@ -0,0 +1,52 @@
+
+Namespace:: Kubernetes namespaces are a way to divide your cluster resources into separate areas that you can deploy apps and restrict access to, such as if you want to share the cluster with multiple teams. For example, system resources that are configured for you are kept in separate namespaces like `kube-system`. If you do not designate a namespace when you create a Kubernetes resource, the resource is automatically created in the `default` namespace.
+
+Service:: A service is a Kubernetes resource that groups a set of pods and provides network connectivity to these pods without exposing the actual private IP address of each pod. You can use a service to make your app available within your cluster or to the public internet.
+
+Deployment:: A deployment is a Kubernetes resource where you can specify information about other resources or capabilities that are required to run your app, such as services, persistent storage, or annotations. You document a deployment in a configuration YAML file, and then apply it to the cluster. The Kubernetes master configures the resources and deploys containers into pods on the compute nodes with available capacity.
++
+Define update strategies for your app, including the number of pods that you want to add during a rolling update and the number of pods that can be unavailable at a time. When you perform a rolling update, the deployment checks whether the update is working and stops the rollout when failures are detected.
++
+A deployment is just one type of workload controller that you can use to manage pods.
+
+Pod:: Every containerized app that is deployed into a cluster is deployed, run, and managed by a Kubernetes resource that is called a pod. Pods represent small deployable units in a Kubernetes cluster and are used to group the containers that must be treated as a single unit. In most cases, each container is deployed in its own pod. However, an app can require a container and other helper containers to be deployed into one pod so that those containers can be addressed by using the same private IP address.
+
+App:: An app can refer to a complete app or a component of an app. You can deploy components of an app in separate pods or separate compute nodes.
+
+[id="rosa-containers_{context}"]
+== Containers
+
+Containers provide a standard way to package your application's code, configurations, and dependencies into a single unit that can run as a resource-isolated process on a compute server. To run your app in Kubernetes on {product-title} ROSA, you must first containerize your app by creating a container image that you store in a container registry. For complete information about Docker, see the link:https://docs.docker.com/[Docker documentation].
```

Comment:
"application's code'

Avoid using `'s` to show possessive form of inanimate objects (IBMSG).

Reply:
"To run your app in Kubernetes on {product-title} ROSA"

^ Was "ROSA" meant to be in parentheses here?

Reply:
Yup. Done.